### PR TITLE
SIGTERM handler

### PR DESCRIPTION
**Pull request by jnippula**
Entrypoint receives SIGTERM (from docker stop) and sends SIGINT to proper process to make ROS2 node to stop gracefully

### CODE REVIEW

**Comment by jnippula** [comment]
Implementation has two cases:
1. Node runs normally while SIGTERM is received
    => the SIGINT sent to node's pid
2. Node has been closed due to some issue (launch process still hanging around)
    => the SIGINT sent to pid 1

example:
Case 1)
```
[rplidar-1] [INFO] [1667544805.101603409] [jari.rplidar]: -------------- Loading parameters --------------
[static_transform_publisher-3] [INFO] [1667544805.102598479] [jari.fcu_to_garmin_static_transform_publisher]: Spinning until killed publishing transform from 'jari/fcu' to 'jari/garmin'
[rplidar-1] [INFO] [1667544805.104132697] [jari.rplidar]: RPLIDAR running on ROS 2 package rplidar_ros. SDK Version: '1.10.0'
Send SIGINT to pid 62
```

Case 2)
```
[static_transform_publisher-3] [INFO] [1667544712.393110939] [jari.fcu_to_garmin_static_transform_publisher]: Spinning until killed publishing transform from 'jari/fcu' to 'jari/garmin'
[ERROR] [rplidar-1]: process has died [pid 62, exit code -6, cmd '/opt/ros/galactic/lib/rplidar_ros2/rplidar --ros-args -r __node:=rplidar -r __ns:=/jari --params-file /opt/ros/galactic/share/rplidar_ros2/config/params.yaml --params-file /tmp/launch_params_bvkxvski -r topic_filtered_out:=~/scan_filtered -r topic_raw_out:=~/scan'].
Pattern not found, send SIGINT to pid 1
```

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+pid=1
+_term() {
+	# FILL UP PROCESS SEARCH PATTERN HERE TO FIND PROPER PROCESS FOR SIGINT:
+	pattern="rplidar_ros2/rplidar"
+
+	pid_value="$(ps -ax | grep $pattern | grep -v grep | awk '{ print $1 }')"
+	if [ "$pid_value" != "" ]; then
+		pid=$pid_value
+		echo "Send SIGINT to pid $pid"
+	else
```
Comment (maseabunikie) [inline entrypoint.sh:35]: Is there missing to set pid=1 ?

Comment (jnippula) [reply entrypoint.sh:35]: It is there above the _term() function

File: entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -e
+
+pid=1
```
Comment (jnippula) [inline entrypoint.sh]: pid=1

File: entrypoint.sh (hunk 1 above)
Comment (jnippula) [reply entrypoint.sh:35]: Of course for readability point of view it could be better to be in else branch, maybe.. or at least inside the function

**Comment by jnippula** [comment]
test

File: entrypoint.sh (hunk 2 above)
Comment (jnippula) [reply entrypoint.sh]: test

**Comment by maseabunikie** [comment]
@jnippula Can you comment on this test message?

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+wait $child
+RESULT=$?
+
+if [ $RESULT -ne 0 ]; then
+		echo "ERROR: RPLidar failed with code $RESULT" >&2
+		exit $RESULT
+else
+		echo "INFO: RPLidar finished successfully, but returning 125 code for docker to restart properly." >&2
+		exit 125
+fi
```
Comment (maseabunikie) [inline entrypoint.sh:66]: @jnippula Can you comment on this second test message?

Comment (jnippula) [reply entrypoint.sh:66]: I think the comments are working now

Comment (jnippula) [reply entrypoint.sh:66]: another test. Previous one had button for single comment. Now I have only Cancel and Comment

Comment (maseabunikie) [reply entrypoint.sh:66]: Seems to work now. :)

**Pull request merged**